Source/SG/SG.cs

```diff
@@ -98,22 +118,43 @@ public static PAENConfiguration GetDefault()
+            rfsgHandle.RF.ExternalGain = -instrConfig.ExternalAttenuation_dBm;
 
-            rfsgHandle.FrequencyReference.Source = RfsgFrequencyReferenceSource.FromString(instrConfig.ReferenceClockSource);
-            
-            if (instrConfig.ShareLOSGToSA)
+            foreach (LocalOscillatorConfiguration loConfig in instrConfig.LoConfigurations)
             {
-                rfsgHandle.RF.LocalOscillator.LOOutEnabled = true;
-                rfsgHandle.RF.LocalOscillator.Source = RfsgLocalOscillatorSource.Onboard;
-            }
-            else
-            {
-                rfsgHandle.RF.LocalOscillator.LOOutEnabled = false;
-                // Return to the default value, in case in future modifications the above case changes
-                // this to something other than the default
-                rfsgHandle.RF.LocalOscillator.Source = RfsgLocalOscillatorSource.Onboard;
+                switch (loConfig.SharingMode)
+                {
+                    case LocalOscillatorSharingMode.None:
+                        rfsgHandle.RF.LOOutExportConfigureFromRfsa = RfsgLOOutExportConfigureFromRfsa.Disabled;
+                        rfsgHandle.RF.LocalOscillator[loConfig.ChannelName].LOOutEnabled = false;
+                        rfsgHandle.RF.LocalOscillator[loConfig.ChannelName].Source = RfsgLocalOscillatorSource.Onboard;
+                        break;
+                    case LocalOscillatorSharingMode.Manual:
+                        rfsgHandle.RF.LOOutExportConfigureFromRfsa = RfsgLOOutExportConfigureFromRfsa.Disabled;
+                        rfsgHandle.RF.LocalOscillator[loConfig.ChannelName].LOOutEnabled = loConfig.ExportEnabled;
+                        rfsgHandle.RF.LocalOscillator[loConfig.ChannelName].Source = RfsgLocalOscillatorSource.FromString(loConfig.Source);
+                        break;
+                    default: // default to automatic case
+                        rfsgHandle.RF.LOOutExportConfigureFromRfsa = RfsgLOOutExportConfigureFromRfsa.Enabled;
+                        rfsgHandle.Utility.ResetAttribute(loConfig.ChannelName, typeof(RfsgChannelBasedLO).GetProperty("LOOutEnabled"));
+                        rfsgHandle.Utility.ResetAttribute(loConfig.ChannelName, typeof(RfsgChannelBasedLO).GetProperty("Source"));
+                        break;
+                }
+
+                switch (loConfig.OffsetMode)
+                {
+                    case LocalOscillatorOffsetMode.NoOffset:
+                        rfsgHandle.RF.Upconverter.FrequencyOffset = 0.0;
+                        break;
+                    case LocalOscillatorOffsetMode.UserDefined:
+                        rfsgHandle.RF.Upconverter.FrequencyOffset = loConfig.Offset_Hz;
+                        break;
+                    default: // default to automatic case
+                        rfsgHandle.Utility.ResetAttribute(typeof(RfsgUpconverter).GetProperty("FrequencyOffset"));
+                        break;

```

Comment:
Aren't all of the properties below dependent on the LO channel? Somehow I think we have to specify which LO channel we mean to set these on or I suspect it will fail on 5830/31.

Reply:
Agreed after in-person dialog